packages/server/lib/socket-base.ts

```diff
@@ -394,7 +395,35 @@ export class SocketBase {
         .then((resp) => {
           return cb({ response: resp })
         }).catch((err) => {
-          return cb({ error: errors.clone(err) })
+          let error = errors.clone(err)
+
+          // https://github.com/cypress-io/cypress/issues/16476
+          // If error is cyclic, remove it, because it causes recursive hasBinary call in socket.io.
+          if (isCircular(error)) {
+            // credits to https://stackoverflow.com/a/11616993/1038927
+            const safeStringify = (obj, indent = 2) => {
+              let cache: any[] | null = []
+              const retVal = JSON.stringify(
+                obj,
+                (key, value) => {
+                  return typeof value === 'object' && value !== null
+                    ? cache!.includes(value)
+                      ? undefined // Duplicate reference found, discard key
+                      : cache!.push(value) && value // Store value in our collection
+                    : value
+                },
+                indent,
+              )
+
+              cache = null
+
+              return retVal
+            }
+
+            error = JSON.parse(safeStringify(error))
```

Comment:
* I didn't create any utility function for this feature because it is used once here. 
* I didn't put this part into `errors.clone()` because I guess we don't need to run `isCircular` every time when there is an error.